### PR TITLE
chore(qwen35): clean up stale docs and dead entry points

### PR DESCRIPTION
**Pull request by wjinxu**
## Summary
Addresses the qwen35 docs / dead-code cleanup from #258:

- Remove dead entry points `probe_model()` + `ModelInfo` and `start_with_model` (zero callers) from `pegainfer-qwen35-4b`.
- Drop the `Status:` enum headers from qwen35 docs and use the current `Last touched` convention.
- Fix obsolete `crates/pegainfer-*` paths to the current top-level layout.
- Reconcile the drifted parity numbers to one ledger: TTFT 234ms / TPOT 11.77ms.
- Lift the retired exact-text e2e gibberish debugging story to `docs/lessons/exact-match-gate-thread-cublas.md`.
- Update `docs/index.md` and the qwen35 roadmap cleanup ledger.

## Test plan
- `cargo fmt --all --check` — pass
- `cargo test --release -p pegainfer-qwen35-4b --test e2e_scheduler -- --nocapture`
  - Current `main` (after #265) is blocked locally by FlashInfer v0.6.12 requiring `cuda::fast_mod_div`, which is not available in the CUDA/CCCL headers on this host.
  - Control run at `c48fe8d` (pre-#265), same host/toolchain/GPU4/weights, passes: `1 passed`, 16.15s.
  - This PR does not touch kernels or runtime execution paths beyond removing zero-call qwen35 entry points.

### CODE REVIEW

File: pegainfer-qwen35-4b/src/lib.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ mod weights;
 use std::path::Path;
 
 use anyhow::{Result, anyhow};
```
[the remark this answers is not in the document]
Comment (wjinxu) [reply pegainfer-qwen35-4b/src/lib.rs:18]: Thanks, but `anyhow!` is still used in `start_engine_with_capacity()` for the device-count and UTF-8 model-path errors, so this import is still required.

**Review by xiaguan — CHANGES_REQUESTED** [review]
Thanks for the cleanup. The code-side removal looks reasonable to me, but I think the documentation needs one more pass before this is ready to merge.

A few parts still describe the retired Qwen3.5 exact-text e2e gate and test_data/Qwen3.5-4B.json as current, while accuracy.md says those are no longer active and the files/tests no longer exist. The branch is also behind current main, so it misses the newer long-context HF golden / GSM8K recovery updates and would reintroduce stale roadmap/index text after rebase.

There is also a small cross-doc mismatch: the Qwen3 roadmap still says the same dead probe_model()+ModelInfo pair exists in qwen35, but this PR removes it.

Overall direction looks good, but I would prefer to rebase on main and make the migrated lesson clearly historical before merging.

**Comment by wjinxu** [comment]
Thanks, addressed.

I rebased on current `main` and preserved the newer #282 scheduler CPU-test roadmap/index updates. I also made the Qwen3.5 exact-text e2e / `test_data/Qwen3.5-4B.json` references clearly historical, kept current accuracy docs pointed at the HF logits gate, and fixed the qwen3 roadmap mismatch about the qwen35 `probe_model()` / `ModelInfo` pair.

Updated branch pushed as `3a23942`.

**Review by xiaguan — APPROVED** [review]
LGTM. The follow-up commit addresses my previous concerns: the stale Qwen3.5 exact-text gate references are now clearly historical/retired, the qwen3 roadmap no longer claims the removed qwen35 probe surface still exists, and the qwen35 dead entry points are gone without leaving callers behind. I also verified fmt, diff check, qwen35 release check, and qwen35 lib tests locally.
